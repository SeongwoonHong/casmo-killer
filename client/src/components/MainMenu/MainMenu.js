import React, { Component } from 'react';
import classnames from 'classnames';
import { NavLink } from 'react-router-dom';
import './MainMenu.scss';

import SubMenu from './SubMenu/SubMenu';

const mainMenuItems = [
  { name: '커뮤니티', link: '/test1', icon: 'home' },
  { name: '맛집', link: '/test2', icon: 'explore' },
  { name: 'Q&A', link: '/test3', icon: 'email' },
  { name: '정보', link: '/test4', icon: 'shopping_cart' }
];

class MainMenu extends Component {
  render() {
    const { layout } = this.props;
    return (
      <nav className={ classnames('side-main-nav', {
        active: layout.isMainMenuVisible
      }) }>
        <div className="side-main-nav-wrapper">
          <div className="side-main-nav-header">
            <a
              className="btn"
              role="button"
              tabIndex={ 0 }
              onClick={ this.props.toggleSearchForm }
              onKeyDown={() => {}}>
              <i className="material-icons">
                search
              </i>
            </a>
            <h1>
              <NavLink to="/">CK BOARD</NavLink>
            </h1>
            <form className={ classnames('search-form-dt', {
              active: layout.isSearchFormVisible && layout.isMainMenuVisible
            }) }>
              <input type="text" placeholder="Search" />
            </form>
          </div>
          <div className={ classnames('side-main-nav-body', {
             multiColumns: layout.isSubMenuVisible
          })}>
            <ul className="main-menu">
              <a
                className="btn"
                role="button"
                tabIndex={ 0 }
                onClick={ this.props.toggleMenu }
                onKeyDown={ () => {} }>
                <i className="material-icons">
                  {
                    layout.isMainMenuVisible ? 'chevron_left' : 'chevron_right'
                  }
                </i>
              </a>
              {
                mainMenuItems.map((menu, index) => {
                  return (
                    <li key={ menu.name }>
                      <a
                        role="button"
                        tabIndex={ 0 }
                        className={ classnames({
                          testing: layout.isSubMenuVisible && index === 0
                        })}
                        onClick={ this.props.toggleSubMenu }
                        onKeyDown={ () => {} }>
                        <i className="material-icons">{ menu.icon }</i>
                        {
                          !layout.isSubMenuVisible || layout.isMainMenuVisible
                            ? <span>{ menu.name }</span>
                            : ''
                        }
                      </a>
                    </li>
                  );
                })
              }
            </ul>
            {
              layout.isSubMenuVisible ? <SubMenu /> : ''
            }
          </div>
        </div>
      </nav>
    );
  }
}

export default MainMenu;
