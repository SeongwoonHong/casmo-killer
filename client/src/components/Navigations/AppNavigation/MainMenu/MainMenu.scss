.Menu-container.Main-menu {
    transform: translateX(-100%);
    .main-menu {
        > li {
            > a {
                border-bottom: none;
                &.active {
                    & + .sub-menu {
                        background-color: #fff;
                        border-bottom: 1px solid $main-menu-border-color;
                        display: block;
                        > li {
                            > a {
                                font-size: 12px;
                                padding: 0 0 0 71px;
                                &.active {
                                    font-weight: 700;
                                    text-decoration: underline;
                                }
                            }
                        }
                    }
                }
            }
            > .sub-menu {
                display: none;
            }
            &:first-child {
                > a {
                    &.active {
                        border-bottom: 1px solid $main-menu-border-color;
                    }
                }
            }
        }
    }
    @media screen and (min-width: 992px) {
        .main-menu {
            > li {
                > a {
                    border-bottom: 2px solid transparent;
                    border-top: none;
                    color: palette(Grey, 500);
                    padding: 0 15px;
                    &.active {
                        background-color: transparent;
                        border-color: palette(Grey, 100);
                        color: palette(Grey, 100);
                        & + .sub-menu {
                            display: none;
                        }
                    }
                    > .material-icons {
                        color: inherit;
                        margin-right: 15px;
                    }
                }
                &:first-child {
                    > a {
                        &.active {
                            border-bottom: 2px solid palette(Grey, 100);
                        }
                    }
                }
            }
        }
    }
}
