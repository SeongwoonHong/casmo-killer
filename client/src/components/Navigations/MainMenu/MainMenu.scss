.Main-menu-overlay {
    @include abs-elem(0, 0, 0, 0, absolute);
    background-color: rgba(0, 0, 0, .7);
    display: none;
    z-index: 88;
    &.active {
        display: block;
    }
    @media screen and (min-width: 1200px) {
        &.active {
            display: none;
        }
    }
}
.Main-menu {
    @include abs-elem(0, auto, auto, auto, fixed);
    background-color: transparent;
    height: 100%;
    transform: translateX(-100%);
    transition: all .3s;
    width: $main-menu-width;
    z-index: 88;
    &.active {
        transform: translateX(0);
    }
    > .main-menu-wrapper {
        height: 100%;
        > .main-menu-header {
            @include flex-box();
        }
        > .main-menu-body {
            background-color: #F5F5F5;
            height: 100%;
            ul {
                > li {
                    float: none;
                    > a {
                        @include flex-box();
                        color: #111111;
                        font-size: 14px;
                        height: $top-nav-size;
                        &:hover {
                            background-color: #EBEBEB;
                        }
                        &.active {
                            font-weight: 700;
                        }
                    }
                }
            }
            > .main-menu {
                > li {
                    > a {
                        border-top: 1px solid transparent;
                        padding: 0 24px;
                        &.active {
                            background-color: #fff;
                            border-color: $main-menu-border-color;
                            & + .sub-menu {
                                background-color: #fff;
                                border-bottom: 1px solid $main-menu-border-color;
                                display: block;
                            }
                            > .material-icons {
                                color: palette(Teal, 700);
                            }
                        }
                        > .material-icons {
                            color: #888888;
                            font-size: 20px;
                            height: $top-nav-size;
                            line-height: $top-nav-size;
                            margin-right: 27px;
                        }
                        > span {
                            line-height: $top-nav-size;
                        }
                    }
                    &:first-child {
                        > a {
                            &.active {
                                border-bottom: 1px solid $main-menu-border-color;
                                border-top: none;
                            }
                        }
                    }
                    > .sub-menu {
                        display: none;
                        > li {
                            > a {
                                font-size: 12px;
                                padding: 0 0 0 71px;
                                &.active {
                                    color: palette(Teal, 900);
                                }
                            }
                        }
                    }
                }
            }
        }
        > .main-menu-footer {
            @include abs-elem(auto, 0, 0, 0);
            border-top: 1px solid $main-menu-border-color;
            color: #000;
            text-align: center;
        }
    }
    @media screen and (min-width: 1200px) {
        top: $top-nav-size;
        > .main-menu-wrapper {
            > .main-menu-header {
                display: none;
            }
        }
    }
}
