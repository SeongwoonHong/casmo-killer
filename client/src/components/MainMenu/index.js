import { connect } from 'react-redux';
import MainMenu from './MainMenu';
import * as actions from '../../actions';

const mapStateToProps = (state) => {
  return {
    menu: state.menu
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    toggleMenu: () => dispatch(actions.toggleMenu())
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(MainMenu);
