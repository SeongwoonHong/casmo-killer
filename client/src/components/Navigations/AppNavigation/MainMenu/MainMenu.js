import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';

import './MainMenu.scss';

import MenuContainer from '../../shared/MenuContainer';

class MainMenu extends Component {

  render() {

    const { menus, active, toggleMenu } = this.props;

    const MenuLinks = (routes) => {
      return routes.map((route) => {
        if (route.name && route.icon) {
          return (
            <li
              key={ route.name }
              className="Main-menu__list__items">
              <NavLink
                className="Main-menu__list__items__links"
                to={ route.path }>
                <i className="Main-menu__list__items__links__icons material-icons">
                  { route.icon }
                </i>
                <span className="Main-menu__list__items__links__text">
                  { route.name }
                </span>
              </NavLink>
              <ul className="Main-menu__list__items__sub sub-menu">
                {
                  route.children
                    ? route.children.map(child => (
                      <li
                        key={ child.name }
                        className="Main-menu__list__items__sub__items">
                        <NavLink
                          className="Main-menu__list__items__sub__items__links"
                          exact
                          to={ route.path + child.path }>
                          { child.name }
                        </NavLink>
                      </li>
                    ))
                    : null
                }
              </ul>
            </li>
          );
        }
        return null;
      });
    };

    return (
      <MenuContainer
        active={ active }
        className="Main-menu"
        onClose={ () => toggleMenu(false) }>
        <ul className="Main-menu__list">
          { MenuLinks(menus) }
        </ul>
      </MenuContainer>
    );

  }

}

export default MainMenu;
