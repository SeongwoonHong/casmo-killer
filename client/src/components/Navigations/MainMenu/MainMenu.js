import React, { Component } from 'react';
import { Route, Switch, NavLink, Link } from 'react-router-dom';
import classnames from 'classnames';

import './MainMenu.scss';

import SearchForm from '../SearchForm/SearchForm';

import { MainMenuRoutes } from '../../../routers';

class MainMenu extends Component {

  render() {

    const { layout } = this.props;

    const MenuLinks = (routes) => {
      return routes.map((route) => {
        if (route.name) {
          return (
            <li key={ route.name }>
              <NavLink
                to={ route.path }>
                <i className="material-icons">
                  { route.icon }
                </i>
                <span>{ route.name }</span>
              </NavLink>
            </li>
          );
        }
        return null;
      });
    };

    const SubMenus = (routes) => {
      return routes.map(route => (
        <Route
          key={ route.name || 'root' }
          exact={ route.exact }
          path={ route.path }
          items={ route.children }
          component={ route.subMenu }
        />
      ));
    };

    return (
      <nav className={ classnames('side-main-nav', {
        toggled: layout.isMainMenuVisible
      }) }>
        <div className="teal darken-3 side-main-nav-wrapper">
          <div className="side-main-nav-header">
            <a
              className="btn"
              role="button"
              tabIndex={ 0 }
              onClick={ this.props.toggleSearchForm }
              onKeyDown={ () => {
              } }>
              <i className="material-icons">
                search
              </i>
            </a>
            <h1>
              <Link
                to="/"
                className="teal-text text-lighten-5">
                CK BOARD
              </Link>
            </h1>
            <SearchForm styleClass={ classnames('mb', {
              active: layout.isSearchFormVisible
            }) } />
          </div>
          <a
            className="btn submenu-toggle"
            role="button"
            tabIndex={ 0 }
            onClick={ this.props.toggleMenu }
            onKeyDown={ () => {
            } }>
            <i className="material-icons">
              {
                layout.isMainMenuVisible
                  ? 'chevron_right'
                  : 'chevron_left'
              }
            </i>
          </a>
          <div className="side-main-nav-body">
            <ul className="main-menu">
              { MenuLinks(MainMenuRoutes) }
            </ul>
            <Switch>
              { SubMenus(MainMenuRoutes) }
            </Switch>
          </div>
        </div>
      </nav>
    );

  }

}

export default MainMenu;
