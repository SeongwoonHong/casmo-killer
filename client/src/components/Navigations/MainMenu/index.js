import { connect } from 'react-redux';
import MainMenu from './MainMenu';
import * as actions from '../../../actions/index';

const mapStateToProps = (state) => {
  return {
    layout: state.layout
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    toggleMenu: () => dispatch(actions.toggleMenu()),
    toggleSearchForm: () => dispatch(actions.toggleSearchForm())
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(MainMenu);
