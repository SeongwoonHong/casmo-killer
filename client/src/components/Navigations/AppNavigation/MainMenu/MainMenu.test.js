/* eslint react/jsx-boolean-value: 0 */
import React from 'react';
import { shallow } from 'enzyme';

import MainMenu from './index';

describe('<MainMenu />', () => {

  let component = null;

  beforeEach(() => {
    component = shallow(
      <MainMenu
        active={ true }
        menus={ [
          {
            name: 'a',
            icon: 'a',
            path: '/'
          },
          {
            name: 'b',
            icon: 'a',
            path: '/'
          },
          {
            name: 'c',
            icon: 'a',
            path: '/'
          }
        ] } />
    );
  });

  it('renders the component consistently', () => {

    expect(component).toMatchSnapshot();
    expect(component.find('.Main-menu__list__items').length).toEqual(3);

  });

});
