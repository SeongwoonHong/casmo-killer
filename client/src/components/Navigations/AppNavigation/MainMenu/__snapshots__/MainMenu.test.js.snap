// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainMenu /> renders the component consistently 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainMenu
    active={true}
    menus={
        Array [
            Object {
              "icon": "a",
              "name": "a",
              "path": "/",
            },
            Object {
              "icon": "a",
              "name": "b",
              "path": "/",
            },
            Object {
              "icon": "a",
              "name": "c",
              "path": "/",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "active": true,
      "children": <ul
        className="Main-menu__list"
>
        <li
                className="Main-menu__list__items"
        >
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="Main-menu__list__items__links"
                        to="/"
                >
                        <i
                                className="Main-menu__list__items__links__icons material-icons"
                        >
                                a
                        </i>
                        <span
                                className="Main-menu__list__items__links__text"
                        >
                                a
                        </span>
                </NavLink>
                <ul
                        className="Main-menu__list__items__sub sub-menu"
                />
        </li>
        <li
                className="Main-menu__list__items"
        >
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="Main-menu__list__items__links"
                        to="/"
                >
                        <i
                                className="Main-menu__list__items__links__icons material-icons"
                        >
                                a
                        </i>
                        <span
                                className="Main-menu__list__items__links__text"
                        >
                                b
                        </span>
                </NavLink>
                <ul
                        className="Main-menu__list__items__sub sub-menu"
                />
        </li>
        <li
                className="Main-menu__list__items"
        >
                <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="Main-menu__list__items__links"
                        to="/"
                >
                        <i
                                className="Main-menu__list__items__links__icons material-icons"
                        >
                                a
                        </i>
                        <span
                                className="Main-menu__list__items__links__text"
                        >
                                c
                        </span>
                </NavLink>
                <ul
                        className="Main-menu__list__items__sub sub-menu"
                />
        </li>
</ul>,
      "className": "Main-menu",
      "onClose": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="Main-menu__list__items"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="Main-menu__list__items__links"
                        to="/"
            >
                        <i
                                    className="Main-menu__list__items__links__icons material-icons"
                        >
                                    a
                        </i>
                        <span
                                    className="Main-menu__list__items__links__text"
                        >
                                    a
                        </span>
            </NavLink>
            <ul
                        className="Main-menu__list__items__sub sub-menu"
            />
</li>,
          <li
            className="Main-menu__list__items"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="Main-menu__list__items__links"
                        to="/"
            >
                        <i
                                    className="Main-menu__list__items__links__icons material-icons"
                        >
                                    a
                        </i>
                        <span
                                    className="Main-menu__list__items__links__text"
                        >
                                    b
                        </span>
            </NavLink>
            <ul
                        className="Main-menu__list__items__sub sub-menu"
            />
</li>,
          <li
            className="Main-menu__list__items"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="Main-menu__list__items__links"
                        to="/"
            >
                        <i
                                    className="Main-menu__list__items__links__icons material-icons"
                        >
                                    a
                        </i>
                        <span
                                    className="Main-menu__list__items__links__text"
                        >
                                    c
                        </span>
            </NavLink>
            <ul
                        className="Main-menu__list__items__sub sub-menu"
            />
</li>,
        ],
        "className": "Main-menu__list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "a",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="Main-menu__list__items__links"
                to="/"
>
                <i
                                className="Main-menu__list__items__links__icons material-icons"
                >
                                a
                </i>
                <span
                                className="Main-menu__list__items__links__text"
                >
                                a
                </span>
</NavLink>,
              <ul
                className="Main-menu__list__items__sub sub-menu"
/>,
            ],
            "className": "Main-menu__list__items",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": Array [
                  <i
                    className="Main-menu__list__items__links__icons material-icons"
>
                    a
</i>,
                  <span
                    className="Main-menu__list__items__links__text"
>
                    a
</span>,
                ],
                "className": "Main-menu__list__items__links",
                "to": "/",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "a",
                    "className": "Main-menu__list__items__links__icons material-icons",
                  },
                  "ref": null,
                  "rendered": "a",
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "a",
                    "className": "Main-menu__list__items__links__text",
                  },
                  "ref": null,
                  "rendered": "a",
                  "type": "span",
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": null,
                "className": "Main-menu__list__items__sub sub-menu",
              },
              "ref": null,
              "rendered": null,
              "type": "ul",
            },
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "b",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="Main-menu__list__items__links"
                to="/"
>
                <i
                                className="Main-menu__list__items__links__icons material-icons"
                >
                                a
                </i>
                <span
                                className="Main-menu__list__items__links__text"
                >
                                b
                </span>
</NavLink>,
              <ul
                className="Main-menu__list__items__sub sub-menu"
/>,
            ],
            "className": "Main-menu__list__items",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": Array [
                  <i
                    className="Main-menu__list__items__links__icons material-icons"
>
                    a
</i>,
                  <span
                    className="Main-menu__list__items__links__text"
>
                    b
</span>,
                ],
                "className": "Main-menu__list__items__links",
                "to": "/",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "a",
                    "className": "Main-menu__list__items__links__icons material-icons",
                  },
                  "ref": null,
                  "rendered": "a",
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "b",
                    "className": "Main-menu__list__items__links__text",
                  },
                  "ref": null,
                  "rendered": "b",
                  "type": "span",
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": null,
                "className": "Main-menu__list__items__sub sub-menu",
              },
              "ref": null,
              "rendered": null,
              "type": "ul",
            },
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "c",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="Main-menu__list__items__links"
                to="/"
>
                <i
                                className="Main-menu__list__items__links__icons material-icons"
                >
                                a
                </i>
                <span
                                className="Main-menu__list__items__links__text"
                >
                                c
                </span>
</NavLink>,
              <ul
                className="Main-menu__list__items__sub sub-menu"
/>,
            ],
            "className": "Main-menu__list__items",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": Array [
                  <i
                    className="Main-menu__list__items__links__icons material-icons"
>
                    a
</i>,
                  <span
                    className="Main-menu__list__items__links__text"
>
                    c
</span>,
                ],
                "className": "Main-menu__list__items__links",
                "to": "/",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "a",
                    "className": "Main-menu__list__items__links__icons material-icons",
                  },
                  "ref": null,
                  "rendered": "a",
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "c",
                    "className": "Main-menu__list__items__links__text",
                  },
                  "ref": null,
                  "rendered": "c",
                  "type": "span",
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": null,
                "className": "Main-menu__list__items__sub sub-menu",
              },
              "ref": null,
              "rendered": null,
              "type": "ul",
            },
          ],
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "active": true,
        "children": <ul
          className="Main-menu__list"
>
          <li
                    className="Main-menu__list__items"
          >
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="Main-menu__list__items__links"
                              to="/"
                    >
                              <i
                                        className="Main-menu__list__items__links__icons material-icons"
                              >
                                        a
                              </i>
                              <span
                                        className="Main-menu__list__items__links__text"
                              >
                                        a
                              </span>
                    </NavLink>
                    <ul
                              className="Main-menu__list__items__sub sub-menu"
                    />
          </li>
          <li
                    className="Main-menu__list__items"
          >
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="Main-menu__list__items__links"
                              to="/"
                    >
                              <i
                                        className="Main-menu__list__items__links__icons material-icons"
                              >
                                        a
                              </i>
                              <span
                                        className="Main-menu__list__items__links__text"
                              >
                                        b
                              </span>
                    </NavLink>
                    <ul
                              className="Main-menu__list__items__sub sub-menu"
                    />
          </li>
          <li
                    className="Main-menu__list__items"
          >
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              className="Main-menu__list__items__links"
                              to="/"
                    >
                              <i
                                        className="Main-menu__list__items__links__icons material-icons"
                              >
                                        a
                              </i>
                              <span
                                        className="Main-menu__list__items__links__text"
                              >
                                        c
                              </span>
                    </NavLink>
                    <ul
                              className="Main-menu__list__items__sub sub-menu"
                    />
          </li>
</ul>,
        "className": "Main-menu",
        "onClose": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="Main-menu__list__items"
>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="Main-menu__list__items__links"
                            to="/"
              >
                            <i
                                          className="Main-menu__list__items__links__icons material-icons"
                            >
                                          a
                            </i>
                            <span
                                          className="Main-menu__list__items__links__text"
                            >
                                          a
                            </span>
              </NavLink>
              <ul
                            className="Main-menu__list__items__sub sub-menu"
              />
</li>,
            <li
              className="Main-menu__list__items"
>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="Main-menu__list__items__links"
                            to="/"
              >
                            <i
                                          className="Main-menu__list__items__links__icons material-icons"
                            >
                                          a
                            </i>
                            <span
                                          className="Main-menu__list__items__links__text"
                            >
                                          b
                            </span>
              </NavLink>
              <ul
                            className="Main-menu__list__items__sub sub-menu"
              />
</li>,
            <li
              className="Main-menu__list__items"
>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            className="Main-menu__list__items__links"
                            to="/"
              >
                            <i
                                          className="Main-menu__list__items__links__icons material-icons"
                            >
                                          a
                            </i>
                            <span
                                          className="Main-menu__list__items__links__text"
                            >
                                          c
                            </span>
              </NavLink>
              <ul
                            className="Main-menu__list__items__sub sub-menu"
              />
</li>,
          ],
          "className": "Main-menu__list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "a",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="Main-menu__list__items__links"
                  to="/"
>
                  <i
                                    className="Main-menu__list__items__links__icons material-icons"
                  >
                                    a
                  </i>
                  <span
                                    className="Main-menu__list__items__links__text"
                  >
                                    a
                  </span>
</NavLink>,
                <ul
                  className="Main-menu__list__items__sub sub-menu"
/>,
              ],
              "className": "Main-menu__list__items",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": Array [
                    <i
                      className="Main-menu__list__items__links__icons material-icons"
>
                      a
</i>,
                    <span
                      className="Main-menu__list__items__links__text"
>
                      a
</span>,
                  ],
                  "className": "Main-menu__list__items__links",
                  "to": "/",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "a",
                      "className": "Main-menu__list__items__links__icons material-icons",
                    },
                    "ref": null,
                    "rendered": "a",
                    "type": "i",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "a",
                      "className": "Main-menu__list__items__links__text",
                    },
                    "ref": null,
                    "rendered": "a",
                    "type": "span",
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": null,
                  "className": "Main-menu__list__items__sub sub-menu",
                },
                "ref": null,
                "rendered": null,
                "type": "ul",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "b",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="Main-menu__list__items__links"
                  to="/"
>
                  <i
                                    className="Main-menu__list__items__links__icons material-icons"
                  >
                                    a
                  </i>
                  <span
                                    className="Main-menu__list__items__links__text"
                  >
                                    b
                  </span>
</NavLink>,
                <ul
                  className="Main-menu__list__items__sub sub-menu"
/>,
              ],
              "className": "Main-menu__list__items",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": Array [
                    <i
                      className="Main-menu__list__items__links__icons material-icons"
>
                      a
</i>,
                    <span
                      className="Main-menu__list__items__links__text"
>
                      b
</span>,
                  ],
                  "className": "Main-menu__list__items__links",
                  "to": "/",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "a",
                      "className": "Main-menu__list__items__links__icons material-icons",
                    },
                    "ref": null,
                    "rendered": "a",
                    "type": "i",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "b",
                      "className": "Main-menu__list__items__links__text",
                    },
                    "ref": null,
                    "rendered": "b",
                    "type": "span",
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": null,
                  "className": "Main-menu__list__items__sub sub-menu",
                },
                "ref": null,
                "rendered": null,
                "type": "ul",
              },
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "c",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  className="Main-menu__list__items__links"
                  to="/"
>
                  <i
                                    className="Main-menu__list__items__links__icons material-icons"
                  >
                                    a
                  </i>
                  <span
                                    className="Main-menu__list__items__links__text"
                  >
                                    c
                  </span>
</NavLink>,
                <ul
                  className="Main-menu__list__items__sub sub-menu"
/>,
              ],
              "className": "Main-menu__list__items",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": Array [
                    <i
                      className="Main-menu__list__items__links__icons material-icons"
>
                      a
</i>,
                    <span
                      className="Main-menu__list__items__links__text"
>
                      c
</span>,
                  ],
                  "className": "Main-menu__list__items__links",
                  "to": "/",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "a",
                      "className": "Main-menu__list__items__links__icons material-icons",
                    },
                    "ref": null,
                    "rendered": "a",
                    "type": "i",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "c",
                      "className": "Main-menu__list__items__links__text",
                    },
                    "ref": null,
                    "rendered": "c",
                    "type": "span",
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": null,
                  "className": "Main-menu__list__items__sub sub-menu",
                },
                "ref": null,
                "rendered": null,
                "type": "ul",
              },
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
